src/main/webapp/app/exam/manage/exam-status.component.ts

```diff
@@ -0,0 +1,187 @@
+    GENERATE_STUDENT_EXAMS = 'generateStudentExams',
+    PREPARE_EXERCISE_START = 'prepareExerciseStart',
+}
+
+export enum ExamReviewState {
+    UNSET = 'unset',
+    PLANNED = 'planned',
+    RUNNING = 'running',
+    FINISHED = 'finished',
+}
+
+export enum ExamConductionState {
+    PLANNED,
+    RUNNING,
+    FINISHED,
+}
+
+@Component({
+    selector: 'jhi-exam-status',
+    templateUrl: './exam-status.component.html',
+    styleUrls: ['./exam-status.component.scss'],
+})
+export class ExamStatusComponent implements OnChanges {
+    @Input()
+    public exam: Exam;
+
+    examConfigurationSteps = [
+        ExamConfigurationStep.CONFIGURE_EXERCISES,
+        ExamConfigurationStep.REGISTER_STUDENTS,
+        ExamConfigurationStep.GENERATE_STUDENT_EXAMS,
+        ExamConfigurationStep.PREPARE_EXERCISE_START,
+    ];
+    configurationStepToFlagMap = new Map<ExamConfigurationStep, boolean>();
+    examChecklist: ExamChecklist;
+    numberOfGeneratedStudentExams: number;
+
+    examPreparationFinished = false;
+    examConductionState: ExamConductionState;
+    examReviewState: ExamReviewState;
+
+    readonly examConfigurationStepEnum = ExamConfigurationStep;
+    readonly examConductionStateEnum = ExamConductionState;
+    readonly examReviewStateEnum = ExamReviewState;
+    readonly round = round;
+    readonly Math = Math;
+
+    // Icons
+    faTimes = faTimes;
+    faTimesCircle = faTimesCircle;
+    faCheckCircle = faCheckCircle;
+    faArrowRight = faArrowRight;
+    faDotCircle = faDotCircle;
+    faCalendarCheck = faCalendarCheck;
+    faCalendarTimes = faCalendarTimes;
+
+    constructor(private examChecklistService: ExamChecklistService) {}
+
+    ngOnChanges(): void {
+        this.examChecklistService.getExamStatistics(this.exam).subscribe((examStats) => {
+            this.examChecklist = examStats;
+            this.numberOfGeneratedStudentExams = this.examChecklist.numberOfGeneratedStudentExams ?? 0;
+
+            // Step 1:
+            this.setExamPreparation();
+        });
+
+        // Step 2: Exam conduction
+        this.setConductionState();
+
+        // Step 3: Exam correction
+        this.setReviewState();
+    }
+
+    /**
+     * Auxiliary method that determines if all configuration steps for the exam exercises are finished
+     * @private
+     * @returns boolean indicating whether configuration is finished
+     */
+    private areAllExercisesConfigured(): boolean {
+        const atLeastOneGroup = this.examChecklistService.checkAtLeastOneExerciseGroup(this.exam);
+
+        const numberOfExercisesEqual = this.examChecklistService.checkNumberOfExerciseGroups(this.exam);
+
+        const noEmptyExerciseGroup = this.examChecklistService.checkEachGroupContainsExercise(this.exam);
+
+        const maximumPointsEqual = this.examChecklistService.checkPointsExercisesEqual(this.exam);
+
+        const examPointsReachable = this.examChecklistService.checkTotalPointsMandatory(maximumPointsEqual, this.exam);
```

Comment:
```suggestion
        const atLeastOneGroup = this.examChecklistService.checkAtLeastOneExerciseGroup(this.exam);
        const numberOfExercisesEqual = this.examChecklistService.checkNumberOfExerciseGroups(this.exam);
        const noEmptyExerciseGroup = this.examChecklistService.checkEachGroupContainsExercise(this.exam);
        const maximumPointsEqual = this.examChecklistService.checkPointsExercisesEqual(this.exam);
        const examPointsReachable = this.examChecklistService.checkTotalPointsMandatory(maximumPointsEqual, this.exam);
```

Thanks for addressing my comment here.
If you would have objected to the suggested change I would have accepted that as well as the balance between enough documentation and too much doc is quite hard and depends a lot on individual preferences and experiences.

> In general I have the feeling that we are a bit economical when it comes to in-depth method documentation. This often leads to code that is hard to maintain and modify for any developer other than the original author.

That is my experience as well. There are definitely a lot of places where the documentation for non-trivial methods should be improved when touching the code.

Reply:
We do not come to a common denominator in terms of code structure, do we? 😄